internal/cli/kraft/cloud/deploy/deploy.go

```diff
@@ -218,6 +225,69 @@ func (opts *DeployOptions) Run(ctx context.Context, args []string) error {
 		return fmt.Errorf("could not prepare deployment: %w", err)
 	}
 
+	if opts.Rollout != "" {
+		paramodel, err := processtree.NewProcessTree(
+			ctx,
+			[]processtree.ProcessTreeOption{
+				processtree.IsParallel(false),
+				processtree.WithRenderer(
+					log.LoggerTypeFromString(config.G[config.KraftKit](ctx).Log.Type) != log.FANCY,
+				),
+				processtree.WithFailFast(true),
+				processtree.WithHideOnSuccess(false),
+				processtree.WithTimeout(opts.Timeout),
+			},
+			processtree.NewProcessTreeItem(
+				"draining",
+				"",
+				func(ctx context.Context) error {
+					instanceClient := kraftcloud.NewInstancesClient(
+						kraftcloud.WithToken(config.GetKraftCloudTokenAuthConfig(*opts.Auth)),
+						kraftcloud.WithDefaultMetro(opts.Metro),
+					)
+
+					var oldInsts []instances.GetResponseItem
+					if utils.IsUUID(opts.Rollout) {
+						oldInsts, err = instanceClient.GetByUUIDs(ctx, opts.Rollout)
+						if err != nil {
+							return fmt.Errorf("could not retrieve old instance, invalid UUID: %w", err)
```

Comment:
The error here won't necessarily be related to an "invalid UUID".

Reply:
makes sense, will edit